### PR TITLE
fix NSSL MP init issue when initialized from other microphysics schemes

### PR DESCRIPTION
**Pull request by JiliDong-NOAA**
This is the fix from Ted Mansell @MicroTed on NSSL MP initialization to check if there is missing number of concentration for any non-zero hydrometers mixing ratio. NSSL MP will calculate N from mixing ratio in the first timestep if missing value is found. This fix will guarantee that NSSL MP enters the correct warm/cold start initializations.

### CODE REVIEW

**Review by Qingfu-Liu — APPROVED** [review]

**Review by grantfirl — APPROVED** [review]

File: physics/MP/NSSL/mp_nssl.F90
Comment (dustinswales) [inline physics/MP/NSSL/mp_nssl.F90]: Can you change "real" to "real(kind_phys)"
(I see a couple other instances in this module as well. Do you mind changing them as well?)

**Review by dustinswales — COMMENTED** [review]
Looks good to me.
I suggest making sure all of the real declarations are parameterized (This helps tremendously for interoperability)

File: physics/MP/NSSL/mp_nssl.F90
Comment (JiliDong-NOAA) [reply physics/MP/NSSL/mp_nssl.F90]: done

**Comment by BrianCurtis-NOAA** [comment]
EPIC wishes to go to the UFSWM PR this is attached to Today/Tomorrow. Since there have been changes since the last review, could we get another review to make sure the recent commits are OK?

**Comment by BrianCurtis-NOAA** [comment]
Forgot to mention, make sure you're ready to sync up to auth repo once the current UFSWM PR is merged later today.

**Review by grantfirl — APPROVED** [review]

**Comment by zach1221** [comment]
@grantfirl testing is complete on UFS-WM PR [#2098](https://github.com/ufs-community/ufs-weather-model/pull/2098) . Could you please merge this PR for us?

**Pull request merged**
